src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -777,10 +859,6 @@ private void ExecuteBlockActionsCore<TBlockStartAction, TBlockAction, TNodeActio
 
             var blockEndActions = PooledHashSet<TBlockAction>.GetInstance();
             var blockActions = PooledHashSet<TBlockAction>.GetInstance();
-            var executableNodeActions = ArrayBuilder<TNodeAction>.GetInstance();
-            var syntaxNodeActions = executableNodeActions as ArrayBuilder<SyntaxNodeAnalyzerAction<TLanguageKindEnum>>;
-            var operationActions = executableNodeActions as ArrayBuilder<OperationAnalyzerAction>;
```

Comment:
these two lines were super wacky.  we'd make the bulder, then decide if we were the codepath that was running for syntax ndoes or for operations.  and hten special case behavior lower down.